apps/aecore/src/aec_keys.erl

```diff
@@ -165,10 +170,28 @@ init([Password, KeysDir]) ->
 %%--------------------------------------------------------------------
 handle_call({sign, _}, _From, #state{priv=undefined} = State) ->
     {reply, {error, key_not_found}, State};
-handle_call({sign, Term}, _From, #state{priv=PrivKey, algo=Algo, digest=Digest, curve=Curve} = State) ->
-    Signature = crypto:sign(Algo, Digest, term_to_binary(Term),  [PrivKey, crypto:ec_curve(Curve)]), %% TODO Review transaction serialization.
-    {reply, {ok, #signed_tx{data = Term, signatures = [Signature]}}, State};
-
+handle_call({sign, Term}, _From,
+            #state{pub = PubKey, priv=PrivKey,
+                   algo=Algo, digest=Digest, curve=Curve} = State) ->
+    Signers = aec_tx:signers(Term),
+    case lists:member(PubKey, Signers) of
+        false ->
+            {reply, {error, not_a_signer}, State};
+        true ->
+            Bin = aec_tx:serialize_to_binary(Term),
+            Signature =
+                crypto:sign(
+                  Algo, Digest, Bin, [PrivKey, crypto:ec_curve(Curve)]),
+            {reply, {ok, #signed_tx{data = Term,
+                                    signatures = [Signature]}}, State}
+    end;
+handle_call({verify, Sigs, Term}, _From, State) ->
+    Signers = aec_tx:signers(Term),
+    Bin = aec_tx:serialize_to_binary(Term),
+    Res = lists:any(fun(Sig) ->
```

Comment:
I understand constraints shall be:
* A. Number of signatures (Sigs) shall be the same as number or signers (Signers);
* B. All signatures shall verify with any one of the signers;
* C. Each signature shall verify with a distinct signer.

Because signatures do not include the pubkey, and because of constraint C, it seems a sequential function would be most readable. (I say this as I do not fully understand this implementation respects these constraints, and I have not reviewed tests yet - that may clarify).

Reply:
Yes, I was pondering whether to make it more strict. This should perhaps be another ticket.

Reply:
Created PT ticket #152714446

Reply:
Created PT ticket #152714484 regarding tracking the implementation of the Tx sync method